tsk/img/img_writer.cpp

```diff
@@ -206,8 +206,22 @@ static TSK_RETVAL_ENUM addNewBlock(TSK_IMG_WRITER* writer, TSK_OFF_T addr, char
     writer->blockToSectorNumber[blockNum] = uint32_t(writer->nextDataOffset / VHD_SECTOR_SIZE);
 
     char * fullBuffer = (char *)tsk_malloc(writer->blockSize * sizeof(char));
+    if (fullBuffer == NULL) {
+        return TSK_ERR;
+    }
+
     char * sectorBitmap = (char *)tsk_malloc(writer->sectorBitmapLength * sizeof(char));
+    if (fullBuffer == NULL) {
```

Comment:
Is this a cut and paste error? You should be testing sectorBitmap I think

Reply:
yes